OpenRA.Game/FileSystem/MixFile.cs

```diff
@@ -66,6 +64,58 @@ public MixFile(FileSystem context, string filename, PackageHashType type, int pr
 			}
 		}
 
+		Dictionary<string, PackageEntry> ParseIndex(Dictionary<uint, PackageEntry> entries)
+		{
+			var classicIndex = new Dictionary<string, PackageEntry>();
+			var crcIndex = new Dictionary<string, PackageEntry>();
+			var allPossibleFilenames = new HashSet<string>();
+
+			// Try and find a local mix database
+			var dbNameClassic = PackageEntry.HashFilename("local mix database.dat", PackageHashType.Classic);
+			var dbNameCRC = PackageEntry.HashFilename("local mix database.dat", PackageHashType.CRC32);
+			foreach (var kv in entries)
+			{
+				if (kv.Key == dbNameClassic || kv.Key == dbNameCRC)
+				{
+					var db = new XccLocalDatabase(GetContent(kv.Value));
+					foreach (var e in db.Entries)
+						allPossibleFilenames.Add(e);
+
+					break;
+				}
+			}
+
+			// Load the global mix database
+			// TODO: This should be passed to the mix file ctor
```

Comment:
Note: This TODO will be addressed in the pr that introduces the concept of a package loader (like our existing sprite/sequence/audio loaders)